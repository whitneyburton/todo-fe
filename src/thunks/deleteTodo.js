import { setLoading, setError, setTodos, setItems } from '../actions';
import { fetchData } from '../utils/api';

export const deleteTodo = (todoId, itemId, items, userId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      await fetchData(`/users/${userId}/todos/${todoId}`, 'DELETE');
      const updatedTodos = await fetchData(`/users/${userId}/todos`, 'GET')
      dispatch(setTodos(updatedTodos));
      const updatedItems = items.filter(item => item.id !== itemId)
      dispatch(setItems(updatedItems))
      return updatedTodos;
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
