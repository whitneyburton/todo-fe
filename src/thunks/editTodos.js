import { setLoading, setError, updateTodo, updateItem } from '../actions';
import { fetchData } from '../utils/api';

export const editTodos = (todoTitle, itemName, userId, todo, item) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      await fetchData(`/users/${userId}/todos/${todo.id}?title=${todoTitle}`, 'PUT');
      const updatedTodo = await fetchData(`/users/${userId}/todos/${todo.id}`, 'GET'); 
      dispatch(updateTodo(updatedTodo));
      
      await fetchData(`/users/${userId}/todos/${todo.id}/items/${item.id}?name=${itemName}`, 'PUT');
      const updatedItem = await fetchData(`/users/${userId}/todos/${todo.id}/items/${item.id}`, 'GET'); 
      dispatch(updateItem(updatedItem));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};