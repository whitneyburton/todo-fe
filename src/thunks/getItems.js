import { setLoading, setError, setItems } from '../actions';
import { fetchData } from '../utils/api';

export const getItems = (todoId = '') => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const response = await fetchData(`/todos/${todoId}/items`, 'GET');
      const items = response.map(item => {
        return { name: item.name, done: item.done, todo_id: item.todo_id }
      })
      dispatch(setItems(items));
      return items;
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
