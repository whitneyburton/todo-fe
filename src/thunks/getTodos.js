import { setLoading, setError, setTodos, setItems } from '../actions';
import { fetchData } from '../utils/api';

export const getTodos = () => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const todos = await fetchData('/todos', 'GET');
      dispatch(setTodos(todos));
      let allItems = [];
      todos.forEach(async todo => {
        const items = await fetchData(`/todos/${todo.id}/items`, 'GET');
        allItems.push(items[0])
        dispatch(setItems(allItems))
      })
      return todos;
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
