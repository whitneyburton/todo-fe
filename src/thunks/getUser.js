import { setLoading, setError, setUser } from '../actions';
import { fetchData } from '../utils/api';

export const getUser = (email, password) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const user = await fetchData(`/sessions?email=${email}&password=${password}`, 'POST');
      dispatch(setUser(user.id));
      return user.id;
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};