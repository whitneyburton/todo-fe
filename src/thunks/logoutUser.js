import { setLoading, setError, setUser } from '../actions';
import { fetchData } from '../utils/api';

export const logoutUser = id => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      await fetchData(`/sessions?id=${id}`, 'DELETE');
      dispatch(setUser(''));
      dispatch(setError(''));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};