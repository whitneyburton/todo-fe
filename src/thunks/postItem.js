import { setError, setLoading, setItem } from '../actions';
import { fetchData } from '../utils/api';

export const postItem = (name, done, todoId, userId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      await fetchData(`/users/${userId}/todos/${todoId}/items?name=${name}&done=${done}`, 'POST');
      const newItem = await fetchData(`/users/${userId}/todos/${todoId}/items`, 'GET'); 
      dispatch(setItem(...newItem));
      return newItem;
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
