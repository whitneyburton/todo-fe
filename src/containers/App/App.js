import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { getTodos } from '../../thunks/getTodos';
import { getItems } from '../../thunks/getItems';
import TodoForm from '../TodoForm/TodoForm';
import TodoContainer from '../TodoContainer/TodoContainer';

export const App = ({ getTodos, getItems }) => {
  useEffect(() => {
    getTodos();
    getItems(16)
  }, []);

  return (
    <div className='App'>
      <h1>Tracker</h1>
      <TodoForm />
      <TodoContainer />
    </div>
  );
};

export const mapDispatchToProps = dispatch => ({
  getTodos: () => dispatch(getTodos()),
  getItems: id => dispatch(getItems(id))
});

export default connect(null, mapDispatchToProps)(App);