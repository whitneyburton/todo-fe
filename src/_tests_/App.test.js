import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapDispatchToProps } from '../containers/App/App';
import { shallow } from 'enzyme';
import { getTodos } from '../thunks/getTodos';
jest.mock('../thunks/getTodos');

describe('App', () => {
  let wrapper;
  let mockError;

  beforeEach(() => {
    mockError = '';

    wrapper = shallow(
      <App
        getEvents={getTodos}
        error={mockError}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
