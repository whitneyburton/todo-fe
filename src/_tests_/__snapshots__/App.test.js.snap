// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="tracker-title"
  >
    TRACKER
  </h1>
  <ConnectFunction />
  <ConnectFunction />
</div>
`;
